src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -694,6 +694,9 @@ private static Task<TResult> RunAsync<TResult>(Func<object, long, TResult> func,
                         lock (s_tasks)
                         {
                             ((ICollection<KeyValuePair<object, Task>>)s_tasks).Remove(new KeyValuePair<object, Task>(key!, task));
+                            // Since it was canceled, func(..) had not executed and call AfterResolution it needs to be called here.
+                            NameResolutionTelemetry.Log.AfterResolution(key!, startingTimestamp);
```

Comment:
Should we add `new OperationCanceledException());` as `exception` argument so Log then states that particular resolution has been canceled?

Reply:
Yes, this should be reported as a failure.
Nit: you shouldn't need the `!` on the key as the nullability is implied by the `!` 1 line above.

Reply:
Instead of newing up a new exception, I think we can use a static instance, since it's only being used to determine `error.type` for the `dns.lookup.duration` metric.

Reply:
> Yes, this should be reported as a failure. Nit: you shouldn't need the `!` on the key as the nullability is implied by the `!` 1 line above.

It is screaming at me without it. In both static analysis and actual compile.

Reply:
> Instead of newing up a new exception, I think we can use a static instance, since it's only being used to determine `error.type` for the `dns.lookup.duration` metric.

I thought about that, however from my point of view, it is not on hot path, and in cancelation path performance is rarely an issue.